src/D2L.Bmx/Program.cs

```diff
@@ -10,11 +10,6 @@
 using D2L.Bmx.Okta;
 using IniParser;
 
-// pre-create objects used by all commands and/or the middleware
-var configProvider = new BmxConfigProvider( new FileIniDataParser() );
-var config = configProvider.GetConfiguration();
-var github = new GitHubClient();
```

Comment:
Loosely related - with the recent changes, these objects are no longer needed for all BMX runs, and creating them here doesn't result in better sharing, but instead may result in unnecessary actions that slow down startup (e.g. reading config on a `bmx --version` is unnecessary)

So now I'm inlining them into the respective command handler and middleware instead.